### PR TITLE
user-can-create-waitlist-entry - Chase Simmonds

### PR DESCRIPTION
**Pull request by chase-simmonds**
https://user-images.githubusercontent.com/65789518/181628316-112e8be7-ee05-48e8-8217-6bef94ceae54.mov

### CODE REVIEW

File: client/components/form.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  const [name, setName] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('');
+  const [barberName, setBarberName] = useState('');
+
+  const submitForm = async event => {
+    event.preventDefault();
+    try {
+      const body = { name, phoneNumber, barberName };
+      // eslint-disable-next-line
```
Comment (thebearingedge) [inline client/components/form.jsx]: I don't think this eslint directive is necessary.

File: client/components/form.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      const response = await fetch('http://localhost:3000/api/waitlist', {
```
Comment (thebearingedge) [inline client/components/form.jsx]: avoid putting in localhost:3000, you should be able to just use `/api/waitlist`.

File: client/components/form.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(body)
+      });
+      window.location = '/';
```
Comment (thebearingedge) [inline client/components/form.jsx]: This shouldn't be necessary. You don't need to reload the universe just to show an extra table row. The fetch-in-react-practice POST feature does what you need.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: client/components/form.jsx (hunk 1 above)
Comment (chase-simmonds) [reply client/components/form.jsx]: I don't call response in the function so it underlines the variable warning me it isn't used

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**